### DISCUSSION TITLE
RAC: Multiple focus rings displayed within field `Group`

### DISCUSSION THREAD

**Discussion by kydecker** [adobe/react-spectrum#7241] [Q&A] [answered]
When navigating with a keyboard, focusing on elements like `Button` within a field `Group` will cause two focus rings to be shown. Here is an example:

![CleanShot 2024-10-23 at 12 45 05@2x](https://github.com/user-attachments/assets/d5ec7852-8b7d-49c4-9655-5b12929fd0b8)

This also happens [within the RAC docs for `Group`](https://react-spectrum.adobe.com/react-aria/Group.html#example) although due to the edge-to-edge styling it's less noticeable:

https://github.com/user-attachments/assets/2fe34687-8a5b-4021-88f6-e879a18160b5

My expectation would be for only one element to display a focus ring at any given time.

Two questions:

1. Is this intended behavior within RAC?
2. If yes, is there a workaround for `Group` to enforce displaying a single focus ring at a time?

**Comment by snowystinger** [comment, marked as answer]
Can you do it by tracking if the button inside has focus? then you can turn off the outer one? The Group only knows that focus is inside of itself, it isn't aware of the kinds of elements in it or if there are multiple focusable elements

It may be helpful to provide a codesandbox demonstrating how you're styling this.

**Reply by kydecker** [reply]
Was able to resolve the issue by doing that. See https://github.com/namesakefyi/namesake/pull/455
